doc/OnlineDocs/developer_reference/deprecation.rst

```diff
@@ -0,0 +1,50 @@
+Deprecation and Removal of Functionality
+========================================
+
+During the course of development, there may be cases where it becomes
+necessary to deprecate or remove functionality from the standard Pyomo
+offering.
+
+Deprecation
+-----------
+
+We offer a set of tools to help with deprecation in
+``pyomo.common.deprecation``.
+
+By policy, when deprecating or moving an existing capability,
+one of the following functions should be imported. In use,
+the ``version`` option should be set to ``TBD``, which
+will be changed as of the next release.
```

Comment:
Policy question: should we start having users just insert the current version (i.e., 6.4.5.dev0) instead of TBD?  We can update the release process to grep for the version number and/or TBD.  I ask because I ran into this with @Robbybp's PR #2727 where he has docstrings that document that an API changed in version 6.4.5 (which is likely never going to exist).  Changing those to TBD would be hard to find.  It might be easier to look for the "devel" version number...